schematool/db/_hive.py

```diff
@@ -0,0 +1,207 @@
+            # to be most compatible with our particular Hive/Hadoop installation.
+            init_file_path = os.path.join(tempfile.gettempdir(), cls.HIVE_INIT_FILENAME)
+            with open(init_file_path, 'w') as f:
+                ran_on = datetime.strftime(datetime.now(), '%Y-%m-%d %H:%M:%S')
+                f.write('0\t%s\t%s\n' % (cls.DUMMY_ALTER_REF, ran_on))
+            os.chmod(init_file_path, 0666)
+
+            try:
+                # Since the file was created on the local filesystem and the Hive client will execute
+                # the following operation such that all local paths are relative to the Hive server's
+                # filesystem, this will only work if the Hive server is running locally.
+                #
+                # Try to load the file anyway - otherwise, warn the user that this step must be done
+                # manually.
+                #
+                # There does not seem to be a better way to bootstrap the history table
+                # without direct access to the host on which the Hive server is running or manipulating
+                # the table via HDFS, which seems outside the scope of this schema tool.
+                cls.execute("LOAD DATA LOCAL INPATH '%s' OVERWRITE INTO TABLE %s" %
+                    (init_file_path, cls.full_table_name))
+            except DbError, e:
+                raise InitError('Unable to initialize history table. If you are pointing to a Hive '
+                    'server running on a remote host, please copy the file \'%s\' to the same path '
+                    'on the Hive server host and re-run the \'init\' command.\n%s' %
+                    (init_file_path, e.message))
+
+        return create_table_result
+
+    @classmethod
+    def conn(cls):
+        """
+        return the hive connection handle to the configured server
+        """
+        config = cls.config
+        try:
+            connection = hive.connect(host=config['host'], port=config.get('port', cls.DEFAULT_PORT),
+                                      authMechanism='NOSASL', user=config['username'],
+                                      password=config['password'])
+        except Exception, e:
+            raise DbError("Cannot connect to Hive Server: %s\n"
+                          "Ensure that the server is running and you can connect normally"
+                          % e.message)
+
+        return connection
+
+    @classmethod
+    def run_file_cmd(cls, filename):
+        """
+        return a 3-tuple of strings containing:
+            the command to run (list)
+            environment variables to be passed to command (dictionary or None)
+            data to be piped into stdin (file-like object or None)
+        """
+        port = cls.config.get('port', cls.DEFAULT_PORT)
+        jdbc_url = 'jdbc:hive2://%s:%s/default;auth=noSasl' % (cls.config['host'], port)
+
+        # Beeline is the recommended command-line client for HiveServer2
```

Comment:
The `beeline` client actually ships with the Hive distribution (version 0.11+). Yeah, it would be a good idea to mention this in the docs. I'll also add a check to assert that the executable exists.